deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1PodHandler.java

```diff
@@ -372,4 +393,74 @@ public boolean cleanUp() {
 	public int getWeight() {
 	    return ResourceKind.POD.getWeight();
 	}
+	
+	public void watch(ApiClient apiClient, String appName, String serviceName, String namespace)
+			throws IOException, ApiException, HyscaleException {
+		OkHttpClient httpClient = apiClient.getHttpClient().newBuilder().readTimeout(0, TimeUnit.SECONDS).build();
+		apiClient.setHttpClient(httpClient);
+		Configuration.setDefaultApiClient(apiClient);
+		CoreV1Api api = new CoreV1Api();
+		Set<String> pods = new HashSet<String>();
+		String selector = ResourceSelectorUtil.getServiceSelector(appName, serviceName);
+		Integer replicas = null;
+		V1DeploymentHandler v1DeploymentHandler = new V1DeploymentHandler();
+		V1Deployment v1Deployment=null;
+		try {
+			v1Deployment = v1DeploymentHandler.get(apiClient, serviceName, namespace);
+			replicas=v1Deployment.getSpec().getReplicas();
+		}catch(Exception e) {
+			replicas=1;
+		}	
+		
+		String latestPodSelector = null;
+		V1DeploymentHandler deploymentHandler = (V1DeploymentHandler) ResourceHandlers
+				.getHandlerOf(ResourceKind.DEPLOYMENT.getKind());
+		latestPodSelector = deploymentHandler.getPodSelector(apiClient, selector, true, namespace);
+		if (!latestPodSelector.contains(POD_TEMPLATE_HASH)) {
+			V1StatefulSetHandler v1StatefulSetHandler = null;
+			v1StatefulSetHandler = (V1StatefulSetHandler) ResourceHandlers
+					.getHandlerOf(ResourceKind.STATEFUL_SET.getKind());
+			latestPodSelector = v1StatefulSetHandler.getPodSelector(apiClient, selector, true, namespace);
+		}
+		if (latestPodSelector != null) {
+			selector = selector + SEPERATOR + SPACE + latestPodSelector;
+		}
```

Comment:
This should have been a simple logic.
1. Find out parent by checking for statefulset and deployment, whatever exists will become the parent ...just return the parent as string like
String getPodParent(appname,service,namespace .....)

2. Call the respective PodParentHandler of getPodSelector -> latestPodSelectorLabel
3. Concatenate the serviceLabeleSelector and latestPodSelectorLabel
4. Use this to watch the pods with the label selector obtained in 3